lib/src/main/java/com/auth0/jwt/algorithms/ECDSAAlgorithm.java

```diff
@@ -5,46 +5,51 @@
 
 import java.security.InvalidKeyException;
 import java.security.NoSuchAlgorithmException;
-import java.security.PrivateKey;
 import java.security.SignatureException;
-import java.security.interfaces.ECKey;
 import java.security.interfaces.ECPrivateKey;
 import java.security.interfaces.ECPublicKey;
 
 class ECDSAAlgorithm extends Algorithm {
 
+    private final ECPublicKey publicKey;
+    private final ECPrivateKey privateKey;
     private final CryptoHelper crypto;
     private final int ecNumberSize;
-    private final ECKey key;
 
-    ECDSAAlgorithm(CryptoHelper crypto, String id, String algorithm, int ecNumberSize, ECKey key) throws IllegalArgumentException {
+    //Visible for testing
+    ECDSAAlgorithm(CryptoHelper crypto, String id, String algorithm, int ecNumberSize, ECPublicKey publicKey, ECPrivateKey privateKey) throws IllegalArgumentException {
         super(id, algorithm);
-        if (key == null) {
-            throw new IllegalArgumentException("The ECKey cannot be null");
+        if (publicKey == null && privateKey == null) {
+            throw new IllegalArgumentException("Both provided Keys cannot be null.");
```

Comment:
I think `IllegalStateException` is better, since the keys are not arguments of this method, they are state of the instance.